### ISSUE TITLE
Add a Manga/Doujin section

### ISSUE THREAD

**Issue by BobHassok** [stashapp/stash#1659]
Clarification: I've been using Manga and Doujin interchangeably.

I would like to see a dedicated manga section. Currently it is not convenient to view manga using the gallery section, as my image galleries and manga galleries mix together and make it hard to sort.

Edit#2: I'm not asking for hentai metadata scraping (that would be a community scraper request anyway) or a bookmark/left off tracker. Basically just a slightly edited copy of gallery, in a new tab.

I would like to see the manga scrollable by clicking the lightbox, as described in #1493 . With a viewing option for "infinite scroll".
(Basically I want a personal version of nhentai.net)

Ideally there would also be an option to specify a manga title when doing a selective scan, but that's not as important right now.

Edit: $20 bounty posted. I didn't know how to leave a comment with the donation that listed the issue number, so I made it the donation name.

**Comment by CuddleBear92** [comment]
Isn't it kinda outside the scope of stash? Also [Lanraragi](https://github.com/Difegue/LANraragi) is an amazing thing that is made for doujinshi usecases. Otherwise i would use Hydrus. 

Would be easier for the Stash Devs to add just a simple button to navigate between the two server pages, between Stash and Lanraragi #1627

**Comment by BobHassok** [comment]
I don't see how it's outside the scope. Stash is a porn organizer and there is plenty of porn manga. I do like the suggestion of LANraragi, I haven't heard of them. However I do have plenty of manga/comics that have both images and video files. Stash supports both, while it appears LANraragi only supports images.

**Comment by silembir** [comment]
What manga/comics have video files?  Wouldn't it then be hentai?  I agree with @CuddleBear92, Lanraragi serves this purpose and Stash currently exists mostly as a porn video manager.

**Comment by Belleyy** [comment]
First thanks for donation (if confirmed) 👍 , but you should have talk to a dev before doing it.

Idea for mix galleries: i think you could tag all your manga with a tag "Manga", then make a Saved Filters with it (you can made it default if you want).

**Comment by CuddleBear92** [comment]
the % of animated content in Doujinshi is really really low. Only a handful of creators does it. It is more common with one off of a few animation clips with CGI archives though compared to doujinshi. 

But really, Stash with all its scrapers and plugins is all based around normal porn and have no metadata types to really support doujinshi correctly or track read states and many many many other things. 
Just adding a section is a ton of work for something that very few would prob use as there is better more focused options for doujinshi.

There is already a ton of options for Doujinshi reading or managing. Lanraragi is prob the one you want for Doujins from like nhentai.... sigh really is the worst site. 
If you want something local with a ton more management options then ComicRack is prob the way... but good luck finding the install exe for that since the website is dead (got a repo with a backup in my name).  Going the CR route would prob need HDoujinshiDownloader and Ubooquity for web hosting/reading. 

Hydrus is the way for ANY loose files for sure. 

But im going off-topic but you sounded in need of some real upgrades.

**Comment by BobHassok** [comment]
It doesn't need to scrape, that would be cool, but is 100% not needed. If Stash is a porn organizer, then that would include manga and hentai. I'm basically just asking for a slightly redesigned gallery section with its own tab. 

@Belleyy Welcome! I actually donated twice. $20 when I first set up Stash and liked it, then this $20 for my request. I thought I was asking for a rather simple addon that could be done whenever, didn't realize I needed to contact a dev first, oops.

**Comment by WithoutPants** [comment]
> Neither the donate button in Stash or the website it takes you to say to talk to a dev before donating.

It's specified in the Bounty tier description on OpenCollective:

![image](https://user-images.githubusercontent.com/53250216/130564632-2ad9222d-d2d0-444c-b6a7-e27bd681221c.png)


> I would like to see a dedicated manga section. Currently it is not convenient to view manga using the gallery section, as my image galleries and manga galleries mix together and make it hard to sort.

As noted by @Belleyy, this requirement can be met with the existing filtering and saved filter functions. The quickest and easiest way to work with this is to tag manga galleries as `manga` and add a saved filter to show only `manga` tagged images/galleries, and set the default filter to exclude `manga` tagged images/galleries if needed. 

Top-level content pages are divided based on media type (video, image, gallery, with scope to add text-based content in the future), with media added automatically to it based on the media type. Specifically, it's _not_ divided based on _content type_. Adding a specific Manga top-level page would be inconsistent with the rest of the system, and also adds further management overhead of moving content between images/galleries and manga.

One possible alternative has been raised in #1297 to allow custom links to pages with a predefined filter, so you'd have a top-level link to manga images, and another for regular images - they would be the same page, but with different filters set.

I've noted your bounty (thanks!), but we'll need to workshop your requirements a bit, since in its current incarnation it's not an appropriate fit for the system.

**Comment by BobHassok** [comment]
@WithoutPants Then I have 1 clarification. Is stash being designed only for real-life porn and not actually all porn?

**Comment by WithoutPants** [comment]
I think the design philosophy allows for any porn generally.

**Comment by ALonelyJuicebox** [comment]
Agreed with what's been stated here. I don't know that a Manga specific section is ideal today, _but_ I do believe the Image/Gallery section could do with some updates to make the viewing of image content more ideal, and I think those updates would readily apply to manga as well for OP's sake.

Here's a link to that issue (and the bounty I donated towards it)
https://github.com/stashapp/stash/issues/1492

**Comment by Telkhine** [comment]
I think something like tag categories or hierarchical tags could be a flexible/generic solution for more types of media. 

If tags had categories, you could do something like this, were the top line is the category and the children are tags:

Media type | Artist | Series | Series Group/Parody | Series Group/Parody
------------ | ------------- | ------------- | ------------- | -------------
Manga | Kentaro Miura | Berserk | Berserk
Artbook | Kentaro Miura | Berserk Warcry Artbook | Berserk
Doujinshi  | John Doe | Berserk X Batman| Berserk | Batman
Western Comic | Bob Kane | Batman: The Man Who Laughs | Batman

These are non-18+ example, obviously, though would equally apply to adult stuff.

I personally use Komga for comics and manga, though I think many people would like to keep 18+ stuff separate. I have many artist CGs that I'd like to better organize in stash, which could benefit from a system like this. I also think if tagging was expanded with categories or a hierarchy, tag descriptions would be needed as mentioned here #1444.

**Comment by magma2017** [comment]
I'd like to add my support for the eventual integration of a reader for manga/doujin/magazine content or just overall improvements to the gallery section to better serve this purpose.

It would be nice if stash eventually served as a one stop shop for any one individual's entire digital collection of adult content.  That's at least how I view what the long-term scope of a project named "Stash" could or should be.  I know personally I've gone from viewing videos to images and back plenty of times.  Needing to open a different app to do these things can be a bit cumbersome in the moment.

Obviously this sort of integration would take some time to do and by no means do I expect it to be high on the priority list for a long time, but it certainly would be welcome.

**Comment by HijackHornet** [comment]
I think it would be great to have an option to divide the image viewer into photo and drawings. Stash is in many ways similar to a booru and it would be great if one could download an image from a booru and add it to its stash without it mixing too much with everything else. https://www.animebox.es/ app allow downloading of pictures and adds their tags into the filename. So its very streamline to bring stuff to stash from it. 
We just need a better option to really have separated both worlds

**Comment by BobHassok** [comment]
@HijackHornet Exactly! I use [grabber](https://github.com/Bionus/imgbrd-grabber) and it will save anywhere I want with whatever tags I want in the filename. So while a scraper would be nice, we could get away with just parsing the filename for a lot of it.

**Comment by Dounial** [comment]
Why not a general hentai-section instead of just mangas/doujin? Then more people could upload different content (like 2D-Hentai, fetishes or/and drawn hentai) and mangas/doujin could be posted to. It would give it more scope and different arts could combine themselves instead of for example using an extra section for hentai.
